core/trino-main/src/main/java/io/trino/type/TypeCoercion.java

```diff
@@ -92,6 +92,30 @@ public boolean isTypeOnlyCoercion(Type source, Type result)
             return sameDecimalSubtype && sameScale && sourcePrecisionIsLessOrEqualToResultPrecision;
         }
 
+        if (source instanceof TimestampWithTimeZoneType sourceTimestampWithTimeZone && result instanceof TimestampWithTimeZoneType targetTimestampWithTimeZone) {
```

Comment:
nit: commit message `Enable coercion for Time types` -> `Enable coercion for temporal types`

Reply:
Is the commit  about "Enabling a coercion" or recognizing more coercions as "type-only"?